frame/balances/module.go

```diff
@@ -41,9 +42,14 @@ func New(index sc.U8, config *Config, logger log.DebugLogger, mdGenerator *primi
 	storage := newStorage()
 
 	module := Module{
-		Index:       index,
-		Config:      config,
-		constants:   constants,
+		Index:     index,
+		Config:    config,
+		constants: constants,
+		mdConstants: &metadataConstants{
```

Comment:
can we either merge metadataConstants with consts struct or if the latter is only needed for metadata maybe just declare it within the Metadata method, instead of having both as module fields?

Reply:
Good idea. I've removed it from the module field and initialized it in the Metadata() method. The **metadataConstants** struct is itself required for building the metadata constants as the function responsible for building it needs to know all the types that participate in the constants metadata of the corresponding module.